File: backend/src/main/java/com/ubre/backend/dto/VehicleIndicatorDto.java
package com.ubre.backend.dto;

import com.ubre.backend.enums.UserStatus;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

// Every vehicle that is visible on the map has some indicator info

@Getter
@Setter
@NoArgsConstructor
public class VehicleIndicatorDto {
    private Long driverId;
    private WaypointDto location;
    private UserStatus status;
    private Boolean panic;

    public VehicleIndicatorDto(Long driverId, WaypointDto location, UserStatus status, Boolean panic) {
        this.driverId = driverId;
        this.location = location;
        this.status = status;
        this.panic = panic;
    }
}

File: frontend/ubre/src/app/dtos/vehicle-indicator-dto.ts
import { UserStatus } from "../enums/user-status";
import { WaypointDto } from "./waypoint-dto";

export class VehicleIndicatorDto {
    public driverId : number;
    public location : WaypointDto;
    public status : UserStatus;
    public panic : boolean = false;

    constructor(driverId : number, location : WaypointDto, userStatus : UserStatus, panic : boolean = false) {
        this.driverId = driverId;
        this.location = location;
        this.status = userStatus;
        this.panic = panic;
    }
}